paasta_tools/paasta_maintenance.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/env python
+# Copyright 2015 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import datetime
+import json
+from socket import getfqdn
+from socket import gethostbyname
+
+from dateutil import tz
+from pytimeparse import timeparse
+from requests import Request
+from requests import Session
+from requests.exceptions import HTTPError
+
+from paasta_tools.mesos_tools import get_mesos_leader
+
+
+def parse_args():
+    """Parses the command line arguments passed to this script"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-d', '--duration',
+        type=parse_timedelta,
+        default='1h',
+        help="Duration of the maintenance window. Any pytimeparse unit is supported.",
+    )
+    parser.add_argument(
+        '-s', '--start',
+        default=now(),
+        help="Time to start the maintenance window. Defaults to now.",
+    )
+    parser.add_argument(
+        'action',
+        choices=['drain', 'undrain', 'down', 'up', 'status'],
+        help="Action to perform on the speicifed hosts",
+    )
+    parser.add_argument(
+        'hostname',
+        nargs='*',
+        default=[getfqdn()],
+        help='Hostname(s) of machine(s) to start draining. '
+        'You can specify <hostname>|<ip> to avoid querying DNS to determine the corresponding IP.',
+    )
+    options = parser.parse_args()
+    return options
+
+
+def base_api():
+    """Helper function for making all API requests"""
+    leader = get_mesos_leader()
+
+    def execute_request(method, endpoint, **kwargs):
+        url = "%s%s" % (leader, endpoint)
+        timeout = 15
+        s = Session()
+        s.auth = load_credentials()
+        req = Request(method, url, **kwargs)
+        prepared = s.prepare_request(req)
+        try:
+            resp = s.send(prepared,
```

Comment:
I believe this is actually valid and works. I'll move `prepared` to its own line, which will allow me to make the overall indentation look a bit more as you were probably expecting.